File: static/js/condiciones_sanitarias_vehiculos.js
$(document).ready(function() {

    setDefaultFechaKardex();

    // Controlar el registro de kardex
    $('#formCondicionesSanitariasVehiculos').on('submit', function(event) {
        event.preventDefault();
        var formElement = document.getElementById('formCondicionesSanitariasVehiculos');
        var formData = new FormData(formElement);

        fetch('/condiciones_sanitarias_vehiculos/', {
            method: 'POST',
            body: formData
        })
        .then(response => response.json())
        .then(data => {
            if (data.status === 'success') {
                Swal.fire({
                    icon: 'success',
                    title: '¡Control creado!',
                    text: 'Se registro el Control de las condiciones sanitarias de los vehículos de transporte para el área seleccionada.',
                    showConfirmButton: false,
                    timer: 1500
                }).then(() => {
                    location.reload();
                });
            } else {
                Swal.fire({
                    icon: 'error',
                    title: 'Error',
                    text: data.message,  // Mostrar el mensaje de error enviado desde el servidor
                });
            }
        })
        .catch(error => {
            console.error('Error en la solicitud:', error);
            Swal.fire({
                icon: 'error',
                title: 'Error',
                text: 'Ocurrió un error inesperado.',
            });
        });
    });
});


function verDetallesCSV(id_header_format, detalle_area, mes, anio) {
    console.log(detalle_area);

    document.getElementById('listaCSV').style.display = 'none';
    document.getElementById('llenarFormularioCSV').style.display = 'block';
    document.getElementById('detallesCSV').style.display = 'block';
    document.getElementById('tituloDetallesCSV').innerText = `Detalles de ${detalle_area} - ${mes}/${anio}`;
    document.getElementById('idCSV_hidden').value = id_header_format;
    document.getElementById('detallearea_hidden').value = detalle_area;
    document.getElementById('mesCSV').value = mes;
    document.getElementById('anioCSV').value = anio;

    fetch(`/condiciones_sanitarias_vehiculos/detalles_condiciones_sanitarias_vehiculos/${id_header_format}`)
        .then(response => response.json())
        .then(data => {
            var tableBody = $('#tablaDetallesCSV');
            tableBody.empty();

            if (Array.isArray(data) && data.length > 0) {
                data.forEach(function(item) {
                    var verificacion_vehiculos = item.verificacion_vehiculos;
                    var estadoColor = item.estado_ac === "PENDIENTE" ? 'color: red;' : 'color: green;';

                    // Crear una fila HTML y dejar en blanco las celdas donde verificación previa sea None para áreas 2 o 3
                    var row = `
                        <tr>
                            <td class="text-center">${item.fecha}</td>
                            <td class="text-center">${item.detalle_motivo_vehiculo}</td>
                            <td class="text-center">${item.documento_referencia}</td>
                            <td class="text-center">${item.total_bultos}</td>
                            <td class="text-center">${item.detalle_tipo_vehiculo}</td>
                            <td class="text-center">${item.num_placa_vehiculo}</td>
                            <td class="text-center">${verificacion_vehiculos[1] === true ? '✅' : (verificacion_vehiculos[1] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[2] === true ? '✅' : (verificacion_vehiculos[2] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[3] === true ? '✅' : (verificacion_vehiculos[3] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[4] === true ? '✅' : (verificacion_vehiculos[4] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[5] === true ? '✅' : (verificacion_vehiculos[5] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[6] === true ? '✅' : (verificacion_vehiculos[6] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[7] === true ? '✅' : (verificacion_vehiculos[7] === false ? '❌' : '')}</td>
                            <td class="text-center">${item.observacion}</td>
                            <td class="text-center" style="${estadoColor}">${item.detalle_accion_correctiva}</td>
                            <td class="text-center">
                                <button type="button" class="btn d-block w-100 mb-1" style="background-color: #FF8C00; color: white;" onclick="modificarEstadoAC(${item.idaccion_correctiva})">
                                    <i class="fas fa-check-circle"></i>
                                </button>
                            </td>
                        </tr>`;
                    tableBody.append(row);
                });
            } else {
                var noDataRow = '<tr><td colspan="20" class="text-center">No hay detalles disponibles para este Control de las condiciones sanitarias de los vehículos de transporte.</td></tr>';
                tableBody.append(noDataRow);
            }
        })
        .catch(error => {
            console.error("Error al cargar los detalles de Control de las condiciones sanitarias de los vehículos de transporte:", error);
            Swal.fire({
                icon: 'error',
                title: 'Error al cargar detalles',
                text: 'Ocurrió un error al cargar los detalles de Control de las condiciones sanitarias de los vehículos de transporte. Inténtalo de nuevo más tarde.',
            });
        });
}

function setDefaultFechaKardex() {
    const today = new Date().toISOString().split('T')[0];  // Obtiene la fecha actual en formato YYYY-MM-DD
    document.getElementById('fecha_CSV').value = today;  // Asigna la fecha al campo de fecha
}

function limpiarCampos() {
    document.getElementById('idCSV_hidden').value = '';
    document.getElementById('fecha_CSV').value = '';
    document.getElementById('motivo_CSV').value = '';
    document.getElementById('documento_referencia').value = '';
    document.getElementById('cantida_bultos').value = '';
    document.getElementById('tipo_vehiculo').value = '';
    document.getElementById('num_placa').value = '';

    document.getElementById('areaCargaHermetica').checked = false;
    document.getElementById('noTransportaPersonas').checked = false;
    document.getElementById('transporteExclusivo').checked = false;
    document.getElementById('pisoTechoLimpio').checked = false;
    document.getElementById('paredesLimpias').checked = false;
    document.getElementById('libreOlores').checked = false;
    document.getElementById('productoProtegido').checked = false;

    document.getElementById('observaciones').value = '';
    document.getElementById('accionesCorrectivas').value = '';

    document.getElementById('detallearea_hidden').value = '';
    document.getElementById('mesCSV').value = '';
    document.getElementById('anioCSV').value = '';
}

function registrarDetalleCSV() {
    var idCSV_hidden = document.getElementById('idCSV_hidden').value;
    var fecha = document.getElementById('fecha_CSV').value;
    var motivo = document.getElementById('motivo_CSV').value;
    var documento_referencia = document.getElementById('documento_referencia').value;
    var cantida_bultos = document.getElementById('cantida_bultos').value;
    var tipo_vehiculo = document.getElementById('tipo_vehiculo').value;
    var num_placa = document.getElementById('num_placa').value;

    var areaCargaHermetica = document.getElementById('areaCargaHermetica').checked ? 'true' : 'false';
    var noTransportaPersonas = document.getElementById('noTransportaPersonas').checked ? 'true' : 'false';
    var transporteExclusivo = document.getElementById('transporteExclusivo').checked ? 'true' : 'false';
    var pisoTechoLimpio = document.getElementById('pisoTechoLimpio').checked ? 'true' : 'false';
    var paredesLimpias = document.getElementById('paredesLimpias').checked ? 'true' : 'false';
    var libreOlores = document.getElementById('libreOlores').checked ? 'true' : 'false';
    var productoProtegido = document.getElementById('productoProtegido').checked ? 'true' : 'false';

    var observaciones = document.getElementById('observaciones').value;
    var accionesCorrectivas = document.getElementById('accionesCorrectivas').value;

    var detallearea = document.getElementById('detallearea_hidden').value;
    var mes = document.getElementById('mesCSV').value; 
    var anio = document.getElementById('anioCSV').value;

    if (!fecha || !motivo || !documento_referencia || !cantida_bultos || !tipo_vehiculo || !num_placa) {
        Swal.fire({
            icon: 'warning',
            title: 'Campos incompletos',
            text: 'Por favor, completa todos los campos obligatorios antes de enviar.',
        });
        return;
    }

    fetch('/condiciones_sanitarias_vehiculos/registrar_condiciones_vehiculos', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({
            idCSV_hidden: idCSV_hidden,
            fecha: fecha,
            motivo: motivo,
            documento_referencia: documento_referencia,
            cantida_bultos: cantida_bultos,
            tipo_vehiculo: tipo_vehiculo,
            num_placa: num_placa,
            areaCargaHermetica: areaCargaHermetica,
            noTransportaPersonas: noTransportaPersonas,
            transporteExclusivo: transporteExclusivo,
            pisoTechoLimpio: pisoTechoLimpio,
            paredesLimpias: paredesLimpias,
            libreOlores: libreOlores,
            productoProtegido: productoProtegido,
            observaciones: observaciones,
            accionesCorrectivas: accionesCorrectivas,
            detallearea: detallearea,
            mes: mes,
            anio: anio
        })
    })
    .then(response => response.json())
    .then(data => {
        if (data.status === 'success') {
            Swal.fire({
                icon: 'success',
                title: 'Agregado',
                text: 'Se agregó exitosamente un registro para el control de las condiciones sanitarias de los vehículos de transporte.',
                showConfirmButton: false,
                timer: 1500
            }).then(() => {
                sessionStorage.setItem('idCSV_hidden', idCSV_hidden);
                sessionStorage.setItem('detallearea', detallearea);
                sessionStorage.setItem('mes', mes);
                sessionStorage.setItem('anio', anio);
                limpiarCampos();
                verDetallesCSV(idCSV_hidden, detallearea, mes, anio);
            });
        } else {
            Swal.fire({
                icon: 'error',
                title: 'Error',
                text: data.message,
            });
        }
    })
    .catch(error => {
        Swal.fire({
            icon: 'error',
            title: 'Error en la solicitud',
            text: 'Ocurrió un error al enviar la solicitud: ' + error,
        });
    });
}

function volverListaCSV() {
    // Ocultar la sección de detalles
    document.getElementById('detallesCSV').style.display = 'none';

    // Mostrar la sección de detalles
    document.getElementById('llenarFormularioCSV').style.display = 'none';

    // Mostrar la lista de todos los kardex
    document.getElementById('listaCSV').style.display = 'block';
}

function modificarEstadoAC(idAC) {
    var idCSV_hidden = document.getElementById('idCSV_hidden').value;
    var detallearea = document.getElementById('detallearea_hidden').value;
    var mes = document.getElementById('mesCSV').value; 
    var anio = document.getElementById('anioCSV').value;

    fetch(`/condiciones_sanitarias_vehiculos/estadoAC/${idAC}`, {
        method: 'POST'
    })
    .then(response => response.json())
    .then(data => {
        if (data.status === 'success') {
            Swal.fire({
                icon: 'success',
                title: 'Registrado',
                text: 'Se registró la corrección de la observación.',
                showConfirmButton: false,
                timer: 1500
            }).then(() => {
                sessionStorage.setItem('idCSV_hidden', idCSV_hidden);
                sessionStorage.setItem('detallearea', detallearea);
                sessionStorage.setItem('mes', mes);
                sessionStorage.setItem('anio', anio);
                verDetallesCSV(idCSV_hidden, detallearea, mes, anio);
            });
        } else {
            Swal.fire({
                icon: 'error',
                title: 'Error',
                text: data.message,
            });
        }
    })
    .catch(error => {
        console.error("Error al modificar el estado de la acción correctiva:", error);
        Swal.fire({
            icon: 'error',
            title: 'Error al modificar estado',
            text: 'Ocurrió un error al modificar el estado de la acción correctiva. Inténtalo de nuevo más tarde.',
        });
    });
}

function finalizarDetallesCSV(id_header_format) {
    Swal.fire({
        title: '¿Estás seguro?',
        text: "No podrás revertir esta acción",
        icon: 'warning',
        showCancelButton: true,
        confirmButtonColor: '#d33',
        cancelButtonColor: '#3085d6',
        confirmButtonText: 'Sí, finalizar',
        cancelButtonText: 'Cancelar'
    }).then((result) => {
        if (result.isConfirmed) {
            // Si el usuario confirma, se procede con el fetch
            fetch('/condiciones_sanitarias_vehiculos/finalizar_Detalles_CSV', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    id_header_format: id_header_format
                })
            })
            .then(response => response.json())
            .then(data => {
                if (data.status === 'success') {
                    Swal.fire({
                        icon: 'success',
                        title: 'Se finalizó',
                        text: 'El estado de la condición ambiental fue modificado a Finalizado.',
                        showConfirmButton: false,
                        timer: 1500
                    }).then(() => {
                        location.reload();
                    });
                } else {
                    Swal.fire({
                        icon: 'error',
                        title: 'Error',
                        text: data.message,
                    });
                }
            })
            .catch(error => {
                Swal.fire({
                    icon: 'error',
                    title: 'Error en la solicitud',
                    text: 'Ocurrió un error al enviar la solicitud: ' + error,
                });
            });
        }
    });
}


function verDetallesCSVFinalizadas(id_header_format, detalle_area, mes, anio) {
    console.log(detalle_area);

    document.getElementById('listaCSV').style.display = 'none';
    document.getElementById('detallesCSV').style.display = 'block';
    document.getElementById('tituloDetallesCSV').innerText = `Detalles de ${detalle_area} - ${mes}/${anio}`;
    document.getElementById('idCSV_hidden').value = id_header_format;
    document.getElementById('detallearea_hidden').value = detalle_area;
    document.getElementById('mesCSV').value = mes;
    document.getElementById('anioCSV').value = anio;

    fetch(`/condiciones_sanitarias_vehiculos/detalles_condiciones_sanitarias_vehiculos/${id_header_format}`)
        .then(response => response.json())
        .then(data => {
            var tableBody = $('#tablaDetallesCSV');
            tableBody.empty();

            if (Array.isArray(data) && data.length > 0) {
                data.forEach(function(item) {
                    var verificacion_vehiculos = item.verificacion_vehiculos;
                    var estadoColor = item.estado_ac === "PENDIENTE" ? 'color: red;' : 'color: green;';

                    // Crear una fila HTML y dejar en blanco las celdas donde verificación previa sea None para áreas 2 o 3
                    var row = `
                        <tr>
                            <td class="text-center">${item.fecha}</td>
                            <td class="text-center">${item.detalle_motivo_vehiculo}</td>
                            <td class="text-center">${item.documento_referencia}</td>
                            <td class="text-center">${item.total_bultos}</td>
                            <td class="text-center">${item.detalle_tipo_vehiculo}</td>
                            <td class="text-center">${item.num_placa_vehiculo}</td>
                            <td class="text-center">${verificacion_vehiculos[1] === true ? '✅' : (verificacion_vehiculos[1] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[2] === true ? '✅' : (verificacion_vehiculos[2] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[3] === true ? '✅' : (verificacion_vehiculos[3] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[4] === true ? '✅' : (verificacion_vehiculos[4] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[5] === true ? '✅' : (verificacion_vehiculos[5] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[6] === true ? '✅' : (verificacion_vehiculos[6] === false ? '❌' : '')}</td>
                            <td class="text-center">${verificacion_vehiculos[7] === true ? '✅' : (verificacion_vehiculos[7] === false ? '❌' : '')}</td>
                            <td class="text-center">${item.observacion}</td>
                            <td class="text-center" style="${estadoColor}">${item.detalle_accion_correctiva}</td>
                            <td class="text-center">
                                <button type="button" class="btn d-block w-100 mb-1" style="background-color: #FF8C00; color: white;" onclick="modificarEstadoAC(${item.idaccion_correctiva})">
                                    <i class="fas fa-check-circle"></i>
                                </button>
                            </td>
                        </tr>`;
                    tableBody.append(row);
                });
            } else {
                var noDataRow = '<tr><td colspan="11" class="text-center">No hay detalles disponibles para este control de condiciones ambientales.</td></tr>';
                tableBody.append(noDataRow);
            }
        })
        .catch(error => {
            console.error("Error al cargar los detalles de control de condiciones ambientales:", error);
            Swal.fire({
                icon: 'error',
                title: 'Error al cargar detalles',
                text: 'Ocurrió un error al cargar los detalles de control de condiciones ambientales. Inténtalo de nuevo más tarde.',
            });
        });
}


//Para filtrar kardex finalizados
function filterCSVCloseArea(anio, mes) {
    // Generar el id esperado del campo de entrada
    const inputId = `filterCACloseArea_${anio}_${mes}`;

    // Obtener el campo de entrada y verificar si existe
    let input = document.getElementById(inputId);
    if (!input) {
        console.error(`No se encontró el campo de entrada con id: ${inputId}`);
        return;
    }

    // Obtener el valor de filtro y convertir a minúsculas
    let filter = input.value.toLowerCase();

    // Generar el id esperado de la tabla
    const tableId = `tableCloseCA_${anio}_${mes}`;

    // Obtener la tabla y verificar si existe
    let table = document.getElementById(tableId);
    if (!table) {
        console.error(`No se encontró la tabla con id: ${tableId}`);
        return;
    }

    // Obtener todas las filas de la tabla
    let tr = table.getElementsByTagName('tr');

    // Recorrer las filas y aplicar el filtro
    for (let i = 1; i < tr.length; i++) {
        let td = tr[i].getElementsByTagName('td')[0];
        if (td) {
            let txtValue = td.textContent || td.innerText;
            tr[i].style.display = txtValue.toLowerCase().includes(filter) ? "" : "none";
        }
    }
}

function filterByDate() {
    const mes = document.getElementById("filtrarCACLOSE").value;

    if (mes) {
        const [anio, mesNum] = mes.split("-");
        const meses = ["Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio", "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"];
        const nombreMes = meses[parseInt(mesNum) - 1];

        // Redirigir a la misma ruta con los parámetros de mes y año
        window.location.href = `?mes=${nombreMes}&anio=${anio}`;
    }
}

//Para filtrar kardex activos
function filterCSVOpenArea() {
    let input = document.getElementById('filtrarAraeCa');
    let filter = input.value.toLowerCase();
    let table = document.getElementById('caTableOpen');
    let tr = table.getElementsByTagName('tr');

    for (let i = 1; i < tr.length; i++) {
        let td = tr[i].getElementsByTagName('td')[0];
        if (td) {
            let txtValue = td.textContent || td.innerText;
            if (txtValue.toLowerCase().indexOf(filter) > -1) {
                tr[i].style.display = "";
            } else {
                tr[i].style.display = "none";
            }
        }
    }
}

File: routes/condiciones_sanitarias_vehiculos.py
from flask import Blueprint, render_template, request, jsonify, send_file
from connection.database import execute_query
from datetime import datetime
from datetime import time
from auth.auth import login_require


condiciones_sanitarias_vehiculos = Blueprint('condiciones_sanitarias_vehiculos', __name__)

@condiciones_sanitarias_vehiculos.route('/', methods=['GET', 'POST'])
@login_require
def condicionesSanitariasVehiculos():
    if request.method == 'GET':
        try:
            # Obtener toas las áreas
            query_areas = "SELECT * FROM areas WHERE idarea = 1 OR idarea = 2 OR idarea = 3 OR idarea = 5;"
            areas = execute_query(query_areas)
            
            #Obtener todos los registros activos de este formatos
            query_creados_CSV = "SELECT id_header_format, detalle_area, mes, anio, fk_idarea, estado FROM v_headers_formats WHERE estado = 'CREADO' AND fk_idtipoformatos = 12;" 
            headers_formats = execute_query(query_creados_CSV)
            
            #Extraer los motivos para mostrarlos en un selector
            motivos = execute_query("SELECT * FROM motivos_sanitarios_vehiculos")
            
            #Extraer los tipos de vehículos para mostrarlos
            vehiculos = execute_query("SELECT * FROM tipos_vehiculos")
            
            #Paginador
            page = request.args.get('page', 1, type=int)
            per_page = 5
            offset = (page - 1) * per_page
            
            # Obtener los parámetros de mes y año desde la URL
            filter_mes = request.args.get('mes', None)
            filter_anio = request.args.get('anio', None)
            
            # Construir condiciones de filtro
            filter_conditions = "WHERE estado = 'CERRADO' AND fk_idtipoformatos = 12"
            if filter_mes and filter_anio:
                # Si hay filtro de mes y año, solo obtenemos registros que coincidan
                filter_conditions += f" AND mes = '{filter_mes}' AND anio = '{filter_anio}'"
                limit_offset_clause = ""  # Sin paginación cuando hay un filtro
            else:
                # Usar paginación si no hay filtro de fecha
                limit_offset_clause = f" LIMIT {per_page} OFFSET {offset}"

            query_csv_finalizados = f"""
                SELECT
                    mes,
                    anio,
                    json_agg(json_build_object(
                        'id_header_format', id_header_format,
                        'mes', mes,
                        'anio', anio,
                        'estado', estado,
                        'fk_idarea', fk_idarea,
                        'detalle_area', detalle_area
                    )) AS registros
                FROM v_headers_formats
                {filter_conditions}
                GROUP BY mes, anio
                ORDER BY 
                    anio::INTEGER DESC,  
                    CASE 
                        WHEN mes = 'Enero' THEN 1
                        WHEN mes = 'Febrero' THEN 2
                        WHEN mes = 'Marzo' THEN 3
                        WHEN mes = 'Abril' THEN 4
                        WHEN mes = 'Mayo' THEN 5
                        WHEN mes = 'Junio' THEN 6
                        WHEN mes = 'Julio' THEN 7
                        WHEN mes = 'Agosto' THEN 8
                        WHEN mes = 'Septiembre' THEN 9
                        WHEN mes = 'Octubre' THEN 10
                        WHEN mes = 'Noviembre' THEN 11
                        WHEN mes = 'Diciembre' THEN 12
                    END DESC
                {limit_offset_clause}
            """

            v_finalizados_CSV = execute_query(query_csv_finalizados)
            
            # Si no hay filtro de fecha, contar el total de páginas
            if not filter_mes and not filter_anio:
                query_count = """SELECT COUNT(*) AS total
                                FROM (SELECT DISTINCT mes, anio 
                                    FROM v_headers_formats 
                                    WHERE estado = 'CERRADO' AND fk_idtipoformatos = 12) AS distinct_months_years;"""
                total_count = execute_query(query_count)[0]['total']
                total_pages = (total_count + per_page - 1) // per_page
            else:
                total_pages = 1  # Solo una "página" si estamos en modo de filtro
            
            return render_template('condiciones_sanitarias_vehiculos.html',
                                    areas=areas,
                                    headers_formats=headers_formats,
                                    motivos=motivos,
                                    vehiculos=vehiculos,
                                    v_finalizados_CSV=v_finalizados_CSV,
                                    page=page,
                                    total_pages=total_pages)
        except Exception as e:
                print(f"Error al obtener datos: {e}")
                return render_template('condiciones_sanitarias_vehiculos.html')

    elif request.method == 'POST':
        try:
            area = request.form.get('selectArea')
            print(area)
            fecha_actual = datetime.now()

            mes_actual = fecha_actual.month
            anio_actual = fecha_actual.year

            mes_consultar = str(mes_actual)
            anio_consultar = str(anio_actual)
            
            consult_eviromental_condition_month =  "SELECT * FROM headers_formats WHERE mes = %s AND anio = %s AND fk_idarea = %s AND estado = 'CREADO' AND fk_idtipoformatos = 12" 
            verificar_producto = execute_query(consult_eviromental_condition_month, (mes_consultar, anio_consultar, area))
            
            if not verificar_producto:
                query_crear_formato = """
                    INSERT INTO headers_formats(mes, anio, estado, fk_idarea, fk_idtipoformatos)
                    VALUES (%s, %s, %s, %s, %s)
                """
                execute_query(query_crear_formato, (mes_actual, anio_actual, 'CREADO', area, 12))

                return jsonify({'status': 'success', 'message': 'Control de las condiciones sanitarias de los vehículos de transporte creado.'}), 200
            else:
                return jsonify({'status': 'error', 'message': 'El formato para esta área ya existe para este mes.'}), 500
        except Exception as e:
            print(f"Error al crear el kardex: {e}")
            return jsonify({'status': 'error', 'message': 'Error al crear el Control de las condiciones sanitarias de los vehículos de transporte'}), 500


@condiciones_sanitarias_vehiculos.route('/detalles_condiciones_sanitarias_vehiculos/<int:id_header_format>', methods=['GET'])
def detalles_condiciones_ambientales(id_header_format):
    # Obtener los detalles de la condición ambiental
    query_detalle_CA = "SELECT * FROM v_detalle_condiciones_vehiculos WHERE fk_id_header_format = %s ORDER BY fecha DESC"
    detalle_CSV = execute_query(query_detalle_CA, (id_header_format,))

    # Obtener las asignaciones de verificación previa para cada detalle
    detalles_formateados = []

    for detalle in detalle_CSV:
        # Formatear la fecha
        detalle['fecha'] = detalle['fecha'].strftime('%d/%m/%Y')  # Formatear la fecha a DD/MM/YYYY

        # Obtener las asignaciones de verificación previa para cada detalle
        query_asignaciones = "SELECT fk_id_verificion_vehiculos FROM asignacion_detalles_condiciones_sanitarias_vehiculos WHERE fk_id_detalle_condicion_sanitaria_vehiculo = %s"
        asignaciones = execute_query(query_asignaciones, (detalle['id_detalle_condicion_sanitaria_vehiculo_transporte'],))

        verificacion = {1: False, 2: False, 3: False, 4: False, 5: False, 6: False, 7: False}
        
        for asig in asignaciones:
            if asig['fk_id_verificion_vehiculos'] in verificacion:
                verificacion[asig['fk_id_verificion_vehiculos']] = True

        # Añadir las asignaciones al detalle
        detalle['verificacion_vehiculos'] = verificacion
        detalles_formateados.append(detalle)

    return jsonify(detalles_formateados)

@condiciones_sanitarias_vehiculos.route('/registrar_condiciones_vehiculos', methods=['POST'])
def registrar_condiciones_vehiculos():
    try:
        data = request.json
        # Extracción de los datos del formulario
        idCSV_hidden = data['idCSV_hidden']
        fecha = data['fecha']
        motivo = data['motivo']
        documento_referencia = data['documento_referencia']
        cantida_bultos = data['cantida_bultos']
        tipo_vehiculo = data['tipo_vehiculo']
        num_placa = data['num_placa']
        
        areaCargaHermetica = data.get('areaCargaHermetica') == 'true'
        noTransportaPersonas = data.get('noTransportaPersonas') == 'true'
        transporteExclusivo = data.get('transporteExclusivo') == 'true'
        pisoTechoLimpio = data.get('pisoTechoLimpio') == 'true'
        paredesLimpias = data.get('paredesLimpias') == 'true'
        libreOlores = data.get('libreOlores') == 'true'
        productoProtegido = data.get('productoProtegido') == 'true'
        
        print(areaCargaHermetica, noTransportaPersonas, transporteExclusivo, pisoTechoLimpio, paredesLimpias, libreOlores, productoProtegido)
        observaciones = data['observaciones'] or None
        accionesCorrectivas = data['accionesCorrectivas'] or None
        
        print(observaciones, accionesCorrectivas)
        idaccion_correctiva = None
        if accionesCorrectivas != None:
            result_accion_correctiva = execute_query("INSERT INTO acciones_correctivas(detalle_accion_correctiva, estado) VALUES (%s,%s) RETURNING idaccion_correctiva", 
                                                    (accionesCorrectivas, 'PENDIENTE'))
            idaccion_correctiva = result_accion_correctiva[0]['idaccion_correctiva']

        # Inserción de detalles_condiciones_sanitarias_vehiculos_transporte
        query_insert_detalle_CSV = """
            INSERT INTO detalles_condiciones_sanitarias_vehiculos_transporte 
            (fecha,
            fk_id_motivo_sanitario_vehiculo,
            documento_referencia,
            total_bultos,
            fk_id_tipo_vehiculo,
            num_placa_vehiculo,
            fk_id_header_format,
            observacion,
            fk_id_accion_correctiva)
            VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s) RETURNING id_detalle_condicion_sanitaria_vehiculo_transporte
        """
        
        result_detalle = execute_query(query_insert_detalle_CSV, 
                                        (fecha, motivo, 
                                        documento_referencia, 
                                        cantida_bultos, 
                                        tipo_vehiculo, 
                                        num_placa, 
                                        idCSV_hidden, 
                                        observaciones, 
                                        idaccion_correctiva))

        id_detalle = result_detalle[0]['id_detalle_condicion_sanitaria_vehiculo_transporte']

        # Inserciones para verificación previa
        if areaCargaHermetica:
            execute_query("INSERT INTO asignacion_detalles_condiciones_sanitarias_vehiculos(fk_id_detalle_condicion_sanitaria_vehiculo, fk_id_verificion_vehiculos) VALUES (%s,%s)", 
                        (id_detalle, 1))
        if noTransportaPersonas:
            execute_query("INSERT INTO asignacion_detalles_condiciones_sanitarias_vehiculos(fk_id_detalle_condicion_sanitaria_vehiculo, fk_id_verificion_vehiculos) VALUES (%s, %s)",
                        (id_detalle, 2))
        if transporteExclusivo:
            execute_query("INSERT INTO asignacion_detalles_condiciones_sanitarias_vehiculos(fk_id_detalle_condicion_sanitaria_vehiculo, fk_id_verificion_vehiculos) VALUES (%s, %s)",
                        (id_detalle, 3))
        if pisoTechoLimpio:
            execute_query("INSERT INTO asignacion_detalles_condiciones_sanitarias_vehiculos(fk_id_detalle_condicion_sanitaria_vehiculo, fk_id_verificion_vehiculos) VALUES (%s, %s)",
                        (id_detalle, 4))
        if paredesLimpias:
            execute_query("INSERT INTO asignacion_detalles_condiciones_sanitarias_vehiculos(fk_id_detalle_condicion_sanitaria_vehiculo, fk_id_verificion_vehiculos) VALUES (%s, %s)",
                        (id_detalle, 5))
        if libreOlores:
            execute_query("INSERT INTO asignacion_detalles_condiciones_sanitarias_vehiculos(fk_id_detalle_condicion_sanitaria_vehiculo, fk_id_verificion_vehiculos) VALUES (%s, %s)",
                        (id_detalle, 6))
        if productoProtegido:
            execute_query("INSERT INTO asignacion_detalles_condiciones_sanitarias_vehiculos(fk_id_detalle_condicion_sanitaria_vehiculo, fk_id_verificion_vehiculos) VALUES (%s, %s)",
                        (id_detalle, 7))

        return jsonify({'status': 'success', 'message': 'Se registró el Control de las condiciones sanitarias de los vehículos de transporte.'}), 200

    except Exception as e:
        print(f"Error al agregar detalle de condiciones ambientales: {e}")
        return jsonify({'status': 'error', 'message': 'Ocurrió un error al registrar el Control de las condiciones sanitarias de los vehículos de transporte.'}), 500


@condiciones_sanitarias_vehiculos.route('/estadoAC/<int:id_ca>', methods=['POST'])
def estadoAC(id_ca):
    try:
        execute_query("UPDATE acciones_correctivas SET estado = %s WHERE idaccion_correctiva = %s", ('SOLUCIONADO', id_ca))
        return jsonify({'status': 'success', 'message': 'Se cambió el estado de esta acción correctiva.'}), 200
    except Exception as e:
        print(f"Error al modificar el estado de la acción correctiva: {e}")
        return jsonify({'status': 'error', 'message': 'No hay acción correctiva para validar.'}), 500

#Finalizar el formato
@condiciones_sanitarias_vehiculos.route('/finalizar_Detalles_CSV', methods=['POST'])
def finalizar_Detalles_CSV():
    try:
        data = request.json
        # Extracción de los datos del formulario
        id_header_format = data['id_header_format']

        execute_query("UPDATE headers_formats SET estado = %s WHERE id_header_format = %s", ('CERRADO', id_header_format))
        return jsonify({'status': 'success', 'message': 'Se finalizo correctamente el formato.'}), 200

    except Exception as e:
        print(f"Error al agregar detalle de kardex: {e}")
        return jsonify({'status': 'error', 'message': 'Ocurrió un error al registrar el formato.'}), 500